File: exercises/101-109-smallproblems/easy7/swapcase.rb

def swapcase(str)
  chars = str.chars
  chars.map! { |char|
    if char.upcase == char
      char.downcase
    else
      char.upcase
    end
  }
  chars.join
end

## Tests

p swapcase('CamelCase') == 'cAMELcASE'
p swapcase('Tonight on XYZ-TV') == 'tONIGHT ON xyz-tv'
p swapcase('Motörhead RULEZ. Not.') == 'mOTÖRHEAD rulez. nOT.'

File: 210/small-problems/text-processing/swap-case.js
function swapcase(string) {
  const LOWERCASE = /[a-z]/;
  const UPPERCASE = /[A-Z]/;

  const swap = function (char) {
    if (LOWERCASE.test(char)) return char.toUpperCase();
    if (UPPERCASE.test(char)) return char.toLowerCase();
    return char;
  };

  return string.split('')
               .map(swap)
               .join('');
}

console.log(swapcase('CamelCase'));         // "cAMELcASE"
console.log(swapcase('Tonight on XYZ-TV')); // "tONIGHT ON xyz-tv"
